src/System.Private.Xml/tests/XmlWriter/CloseOnAsync.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace System.Xml.Tests
+{
+    public class CloseOnAsync
+    {
+        public class ExtendedMemoryStream : MemoryStream
+        {
+            public bool FlushCalled { get; private set; }
+            public bool FlushAsyncCalled { get; private set; }
+
+            public override void Flush()
+            {
+                FlushCalled = true;
+                base.Flush();
+            }
+
+            public override Task FlushAsync(CancellationToken cancellationToken)
+            {
+                FlushAsyncCalled = true;
+                return base.FlushAsync(cancellationToken);
+            }
+        }
+
+        [Fact]
+        public void AsyncWriter_DoesNotCall_Flush_On_Close()
+        {
+            var stream = new ExtendedMemoryStream();
+            var writer = XmlWriter.Create(stream, new XmlWriterSettings { Async = true });
+            writer.Close();
+
+            Assert.False(stream.FlushCalled);
+        }
+
+        [Fact]
+        public void AsyncWriter_DoesCall_FlushAsync_On_Close()
+        {
+            var stream = new ExtendedMemoryStream();
+            var writer = XmlWriter.Create(stream, new XmlWriterSettings { Async = true });
```

Comment:
```suggestion
            XmlWriter writer = XmlWriter.Create(stream, new XmlWriterSettings { Async = true });
```

Reply:
Why would you declare it explicitly as it is implied from the name and the statement, plus it's a small method.

Reply:
@MihaMarkic, That's the corefx style-guide. It uses `var` when the type is clearly obvious from the right hand side. Clearly obvious means that it's determined by a language rule, like creating on object with: `Type type = new Type(..)` or casting.

Reply:
The style allows var pretty only for "new" or an explicit cast
https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

Reply:
OK, thanks guys